runtime/ms-rest-azure/lib/azureServiceClient.js

```diff
@@ -31,320 +31,320 @@ var moduleVersion = packageJson.version;
-    this.generateClientRequestId = options.generateClientRequestId;
-  }
+    super(credentials, options);
 
-  if (options.longRunningOperationRetryTimeout !== null && options.longRunningOperationRetryTimeout !== undefined) {
-    this.longRunningOperationRetryTimeout = options.longRunningOperationRetryTimeout;
-  }
+    this.acceptLanguage = 'en-US';
+    this.generateClientRequestId = true;
+    this.longRunningOperationRetryTimeout = 30;
 
-  this.addUserAgentInfo(util.format('%s/%s', moduleName, moduleVersion));
-}
+    if (!options) options = {};
 
-util.inherits(AzureServiceClient, msRest.ServiceClient);
+    if (options.acceptLanguage !== null && options.acceptLanguage !== undefined) {
+      this.acceptLanguage = options.acceptLanguage;
+    }
 
-/**
- * Poll Azure long running PUT or PATCH operation. (Deprecated, new version of the code-gen will generate code to call getLongRunningOperationResult)
- * @param {object} [resultOfInitialRequest] - Response of the initial request for the long running operation.
- * @param {object} [options]
- * @param {object} [options.customHeaders] headers that will be added to request
- */
-AzureServiceClient.prototype.getPutOrPatchOperationResult = function (resultOfInitialRequest, options, callback) {
-  return this.getLongRunningOperationResult(resultOfInitialRequest, options, callback);
-};
+    if (options.generateClientRequestId !== null && options.generateClientRequestId !== undefined) {
+      this.generateClientRequestId = options.generateClientRequestId;
+    }
 
-/**
- * Poll Azure long running POST or DELETE operations. (Deprecated, new version of the code-gen will generate code to call getLongRunningOperationResult)
- * @param {object} [resultOfInitialRequest] - result of the initial request.
- * @param {object} [options]
- * @param {object} [options.customHeaders] headers that will be added to request
- */
-AzureServiceClient.prototype.getPostOrDeleteOperationResult = function (resultOfInitialRequest, options, callback) {
-  return this.getLongRunningOperationResult(resultOfInitialRequest, options, callback);
-};
+    if (options.longRunningOperationRetryTimeout !== null && options.longRunningOperationRetryTimeout !== undefined) {
+      this.longRunningOperationRetryTimeout = options.longRunningOperationRetryTimeout;
+    }
 
-/**
- * Poll Azure long running PUT, PATCH, POST or DELETE operations.
- * @param {object} [resultOfInitialRequest] - result of the initial request.
- * @param {object} [options]
- * @param {object} [options.customHeaders] headers that will be added to request
- */
-AzureServiceClient.prototype.getLongRunningOperationResult = function (resultOfInitialRequest, options, callback) {
-  var self = this;
-  
-  if (!callback && typeof options === 'function') {
-    callback = options;
-    options = null;
-  }
-  if (!callback) {
-    throw new Error('Missing callback');
+    this.addUserAgentInfo(util.format('%s/%s', moduleName, moduleVersion));
```

Comment:
Since you're using ES6 throughout, this is a good candidate too. Use `interpolated` or `template` strings and even see if we can lose the dependency on `util`